syft/pkg/cataloger/golang/utils.go

```diff
@@ -0,0 +1,198 @@
+package golang
+
+import (
+	"bytes"
+	"debug/elf"
+	"debug/macho"
+	"debug/pe"
+	"fmt"
+	"io"
+)
+
+// An exe is a generic interface to an OS executable (ELF, Mach-O, PE, XCOFF).
+// TODO: add back close method since we have a closer
+type exe interface {
```

Comment:
`Close()` was removed from here, however, with the `io.ReadCloser` being passed in it doesn't look like omitting this function is necessary.